tests/migration/prepare_addons_for_migration.pm

```diff
@@ -0,0 +1,27 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Make sure all addons are registered then remove -release packages from dropped modules before installation
+
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base 'y2_installbase';
+use strict;
+use warnings;
+use testapi;
+use migration 'remove_dropped_module_packages';
+use registration 'register_addons_cmd';
+
+
+sub run {
+    unless (script_run("SUSEConnect --status-text |grep -a3 basesystem |grep 'Not Registered'")) {
+        assert_script_run("SUSEConnect --cleanup");
+        assert_script_run("SUSEConnect -p sle-module-basesystem/15.3/x86_64");
```

Comment:
we should use the library to avoid writing code for specific arch.

Reply:
I gave up on this, whatever we do, at some point the system is de-registered entirely and the registration code expires, there is no workaround unless we change that behgavior on SCC server's side: more or less after 4pm no maintenance job from previous day can be re-launched. I removed the workaround as it does not work anyway.